### ISSUE TITLE
Tilemap created hundreds of tiles on Godot 4 migration

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot#71188]
### Godot version

4.0 beta 10

### System information

Linux

### Issue description

I have a single tileset with multiple tiles, in Godot 3.5 there were hundreds of tiles but all selectable from the same tileset view.

Now, after migrating to Godot 4 it has recreated the same 400 tiles but you cannot select an item from the tileset image on the right to draw. The only item selectable from the image is the currently selected tile. The only way to do it is to find which one in your list of 400 or so items in the list on the left you can see (which as you can see is not even showing the tile, but the entire altas) or to find an existing tile on the map, grab it then draw. 

I would expect the migration to create a single tile (in new tilemap terms) with all the items available in the base tileset. As it is, tilemaps just cannot be migrated to Godot 4.

![image](https://user-images.githubusercontent.com/12863685/211670023-f43c663b-fd50-4f39-b757-8194f68789b7.png)

### Steps to reproduce

as description

### Minimal reproduction project

as description

**Comment by groud** [comment]
There's kind of no way around it. If you defined your tiles as single tiles, which can have completely various shape, then the new system cannot really group them in a single sheet within a fixed-size grid. Atlas tiles (in 3.x terms) should import fine though.

I guess some logic could be improved here, so that tiles end up grouped in the same atlas when possible, but that logic might be complex to implement, I am not sure. If you want, you may try to merge the atlases together by selecting all of them, then right-click -> merges atlas sources.

**Comment by chucklepie** [comment]
No, what I mean is in Godot 3.5 there was just one tileset and each item you see in the sheet was defined as a tile. On importing to Godot 4 it literally only has one tile as highlighted in image below, but it is repeated 400 times instead of creating same as in godot 3.5 which is one tile for each cell.

btw, I don't know what you mean by merging tiles, but there is no right-click option in either the tileset or the tilemap view on either the sheet or the list of tiles.

![image](https://user-images.githubusercontent.com/12863685/211899565-22d81499-8ef3-4704-a330-58ffc4c09780.png)

**Comment by groud** [comment]
> No, what I mean is in Godot 3.5 there was just one tileset and each item you see in the sheet was defined as a tile.

Yeah, I understood correctly. The problem with the system in 4.0 is that each tile needs to be of a size which is a multiple of the grid's tile size, and aligned to this grid. This was not required in 3.x. As a consequence, the simplest way to ensure tiles were "correctly" imported (as they would still display like they used to) was to create an atlas per tile. It's not optimal, but it's was kind of simpler than trying to infer a grid size out of a list of possibly random rects in an atlas.

I think this could be improved somehow for common use cases, but ultimately there won't be a perfect backward compatible tool to solve this issues, as 4.x atlases have a bit stricter requirements than 3.x.

> btw, I don't know what you mean by merging tiles, but there is no right-click option in either the tileset or the tilemap view on either the sheet or the list of tiles.

Ah sorry, it's in this menu:
![atlas_merging](https://user-images.githubusercontent.com/6093119/211902818-dcef4112-2784-49a9-be3c-17d69a4b5fbe.png)

**Comment by chucklepie** [comment]
I then tried your suggestion of merging atlases, I did a very simple test of a new tilemap with two 'tiles' using the same image but created one tile in one and one in the other. The merge looked ok, but the result was this:

![image](https://user-images.githubusercontent.com/12863685/211903626-937a7a68-68ab-4aa9-b8aa-05b7251a0743.png)


btw, is there any way to get a normal grid instead of this?

![image](https://user-images.githubusercontent.com/12863685/211903792-3eb7c336-27ab-4981-b570-f52eeb55f0d1.png)

**Comment by chucklepie** [comment]
btw, two random questions. 
1. If it can be guaranteed that tile sizes are equal and valid according to grid, I wonder if a conversion option could be supplied as I expect most people migrating old projects will be using 2d tilemaps?

2. if you are calling the list on the left the 'tiles', but it really represents what was the 'tileset' in godot 3.5, what are you calling each individual cell you create which in normal tilemaps/tilesets are tiles? And there appears to be no 'view  id' for the tiles/cells (for when you want to code set_cell, etc

![image](https://user-images.githubusercontent.com/12863685/211904967-08c90193-d627-4cb4-aa34-77175f8a3978.png)

**Comment by groud** [comment]
> I then tried your suggestion of merging atlases, I did a very simple test of a new tilemap with two 'tiles' using the same image but created one tile in one and one in the other. The merge looked ok.

That's a bug it seems, could use it's own report I believe.

> btw, is there any way to get a normal grid instead of this?

Hmm, if two tiles are in the same tilemap, I believe the grid should cover the whole space between the two tiles (like, as a rectangular area). The right tile on the right must belong to another TileMap I guess. 
We cannot draw the whole grid as with bigger unzoom values this would detroy the editor performances. So it limits itself to the area with tiles inside.

> If it can be guaranteed that tile sizes are equal and valid according to grid, I wonder if a conversion option could be supplied as I expect most people migrating old projects will be using 2d tilemaps?

A conversion option isn't really possible, as the conversion process does not really have an interface to interact with. I guess that, however, the conversion process could try to infer a grid size whenever that's possible. But that might be tricky to implement.

> if you are calling the list on the left the 'tiles

The list on the right list "TileSet sources". Usually you need an TileSetAltasSource per texture, but the import generate more than that due to the limitation I mentioned above.

**Comment by chucklepie** [comment]
Ok. I think perhaps for a game with tens of maps with thousands of tiles I'd better stick with 3.5 :(

But can you answer this please, there seems to be no means of viewing the tileset/atlas/source ID in the tileset/tilemap pane like you could with 3.x to know what the IDs are for when calling set/get cell, etc. and in the main screen (as below) this ability to see what you are looking at has gone too?

![image](https://user-images.githubusercontent.com/12863685/211906244-8051d59b-d04a-48fb-995b-185f96a3e905.png)

**Comment by chucklepie** [comment]
btw, to fix this 'grid' issue (as to me it's really offputting not being able to see the map with a grid) you can use the 'snap' option to recreate a grid of any size. I do this in 3.x to have a screen sized grid as well as the normal 32x32 grid.

Given snap is not much use, could you not add a 'grid' option to the toolbar to draw a custom grid on screen?

**Comment by Calinou** [comment]
MRPs from @Poobslag:

[big-tilemap-godot3.zip](https://github.com/godotengine/godot/files/11635356/big-tilemap-godot3.zip) (Godot 3 version of the project)

[big-tilemap-godot4.zip](https://github.com/godotengine/godot/files/11635357/big-tilemap-godot4.zip) (Godot 4 version of the project after the auto-conversion project)
